### ISSUE TITLE
 Stride of TorchSparse Conv3d is no effect?

### ISSUE THREAD

**Issue by haiduo** [mit-han-lab/torchsparse#168]
Hello, 
 I have a problem about Stride of TorchSparse Conv3d. Through experiments, I verify that stride(2) does not have scale influence on the output coordinates. In other words, STRIDE does not work in TrochSparse. For example, spnn.Conv3d(32, 64, 2, stride=2)
![image](https://user-images.githubusercontent.com/35062202/189078165-4eef56ba-6ebc-4edd-a4b7-8af5145a247d.png)
look 
![image](https://user-images.githubusercontent.com/35062202/189077999-00c46d92-977f-49a6-a215-9ddd1f4e8c6c.png)
In addition, it is not the correct coordinate range output of subsample even set the k2s2. As mentioned before [Conv3d with kernel_size=1 and stride=2 doesn't subsample the input tensor](https://github.com/mit-han-lab/torchsparse/issues/23) and [Why not constraining the max coordinates during downsampling?](https://github.com/mit-han-lab/torchsparse/issues/101)
So, at the moment, I suspect that stride > 1 for spnnConv3d is not valid.

**Comment by kentang-mit** [comment]
Hi @haiduo,

In our implementation actually stride > 1 changes the **resolution** of the coordinates, instead of the **scale** of the coordinates. After n downsampling layers, all the coordinates will be a multiple of $2^n$. Note that this is equivalent to reducing the coordinate range by 2$\times$ at each downsampling layer.

Best,
Haotian

**Comment by haiduo** [comment]
> Hi @haiduo,
> 
> In our implementation actually stride > 1 changes the **resolution** of the coordinates, instead of the **scale** of the coordinates. After n downsampling layers, all the coordinates will be a multiple of 2n. Note that this is equivalent to reducing the coordinate range by 2x at each downsampling layer.
> 
> Best, Haotian

Thank you for your reply!
I know mean what you said. Maybe I used an error word about "scale", but I want to say means what you said. That is stride > 1 changes the **resolution** of the coordinates, but I found is not correct about the output resolution of coordinates through experiments.
Specifically, I should use TorchSparse to replace the 3DConv layer of VoxelNet. After two times of downsampling (stride=2), it was found that the coordinate resolution did not reduce the corresponding multiple. Because there needs to be a Voxel2dense operation (for example, Torch. sparse_COO_tensor) for subsequent RPN processing. But I found that the resolution of the generated Dense tensor was not downsampled relative to the input coordinates.
I don't know if I made myself clear.
Finally, thank you again for your reply.
